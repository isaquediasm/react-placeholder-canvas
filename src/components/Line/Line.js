import React from 'react';
import PropTypes from 'prop-types';
import { invertIntervals, renderIntervals } from '../../helpers';
import { StyledOffset } from './Line.styled';

function Component({ style = {}, height, intervals }) {
  return (
    <div style={{ height, ...style }}>
      {renderIntervals(invertIntervals(intervals)).map((styling, idx) => (
        <StyledOffset
          key={idx}
          className='placeholder-offset'
          style={styling}
        />
      ))}
    </div>
  );
}

const Line = React.memo(Component);

Line.propTypes = {
  height: PropTypes.number.isRequired,
  intervals: PropTypes.array,
};

export default Line;
