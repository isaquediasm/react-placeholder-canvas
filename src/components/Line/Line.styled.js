import styled from 'styled-components';

const StyledOffset = styled.div`
  position: absolute;
  display: inline-block;
  background-color: #fff;
  height: inherit;
`;

export { StyledOffset };
