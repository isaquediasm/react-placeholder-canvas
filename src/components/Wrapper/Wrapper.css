@-webkit-keyframes placeHolderShimmer {
  0% {
    background-position: -600px 0;
  }

  100% {
    background-position: 600px 0;
  }
}

@-moz-keyframes placeHolderShimmer {
  0% {
    background-position: -600px 0;
  }

  100% {
    background-position: 600px 0;
  }
}

@keyframes placeHolderShimmer {
  0% {
    background-position: -600px 0;
  }

  100% {
    background-position: 600px 0;
  }
}

.placeholder-wrapper {
  -webkit-animation: placeHolderShimmer 1s linear infinite forwards;
  -moz-animation: placeHolderShimmer 1s linear infinite forwards;
  animation: placeHolderShimmer 1s linear infinite forwards;
  background: linear-gradient(to right, #d2d2d2 8%, #ababab 18%, #d2d2d2 33%);
  background-size: 1200px 104px;
  height: 100%;
  position: relative;
}

.placeholder-line {
  height: 18px;
}

.placeholder-offset {
  position: absolute;
  display: inline-block;
  background-color: #fff;
  height: inherit;
}

.placeholder-container {
  height: inherit;
}

.feed .placeholder-wrapper {
  border-top: 1px solid #555;
  border-bottom: 1px solid #555;
}

.feed .placeholder-wrapper:first-child {
  border-top: none;
}