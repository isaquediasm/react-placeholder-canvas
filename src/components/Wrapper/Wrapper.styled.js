import styled, { keyframes } from 'styled-components';

const shimmer = keyframes`
  0% {
    background-position: -600px 0;
  }

  100% {
    background-position: 600px 0;
  }
`;

const StyledWrapper = styled.div`
  height: 100%;
  position: relative;
  animation: ${shimmer} 1s linear infinite forwards;
  background: linear-gradient(to right, #d2d2d2 8%, #ababab 18%, #d2d2d2 33%);
  background-size: 1200px 104px;
`;

export { StyledWrapper };
