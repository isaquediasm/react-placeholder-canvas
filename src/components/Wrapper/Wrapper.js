import React from 'react';
import PropTypes from 'prop-types';
import { calculateHeight } from '../../helpers';
import { StyledWrapper } from './Wrapper.styled';

function Component({
  style = {},
  children,
  height = calculateHeight(children),
}) {
  return <StyledWrapper style={{ height, ...style }}>{children}</StyledWrapper>;
}

const Wrapper = React.memo(Component);

Wrapper.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element),
};

export default Wrapper;
