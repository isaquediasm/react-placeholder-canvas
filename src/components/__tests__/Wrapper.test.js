import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../Wrapper/Wrapper';
import Line from '../Line/Line';

it('should render lines correctly', () => {
  const tree = renderer
    .create(
      <Wrapper>
        <Line height={10} intervals={[[0, 10], [20, 30]]} />
        <Line height={10} />
        <Line height={10} intervals={[[0, 10], [20, 30]]} />
      </Wrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
