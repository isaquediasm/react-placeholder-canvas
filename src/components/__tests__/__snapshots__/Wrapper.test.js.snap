// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render lines correctly 1`] = `
<div
  className="rxv491-0 hUCnVU"
  style={
    Object {
      "height": 30,
    }
  }
>
  <div
    style={
      Object {
        "height": 10,
      }
    }
  >
    <div
      className="sc-1hs1toa-0 ekigly placeholder-offset"
      style={
        Object {
          "left": "10%",
          "width": "10%",
        }
      }
    />
    <div
      className="sc-1hs1toa-0 ekigly placeholder-offset"
      style={
        Object {
          "left": "30%",
          "width": "70%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "height": 10,
      }
    }
  >
    <div
      className="sc-1hs1toa-0 ekigly placeholder-offset"
      style={
        Object {
          "left": "0%",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "height": 10,
      }
    }
  >
    <div
      className="sc-1hs1toa-0 ekigly placeholder-offset"
      style={
        Object {
          "left": "10%",
          "width": "10%",
        }
      }
    />
    <div
      className="sc-1hs1toa-0 ekigly placeholder-offset"
      style={
        Object {
          "left": "30%",
          "width": "70%",
        }
      }
    />
  </div>
</div>
`;
